core/src/main/java/com/cloud/info/ConsoleProxyInfo.java

```diff
@@ -56,14 +56,12 @@ public ConsoleProxyInfo(boolean sslEnabled, String proxyIpAddress, int port, int
     private String formatProxyAddress(String consoleProxyUrlDomain, String proxyIpAddress) {
         StringBuffer sb = new StringBuffer();
         // Domain in format *.example.com, proxy IP is 1.2.3.4 --> 1-2-3-4.example.com
-        if (consoleProxyUrlDomain.startsWith("*")) {
+        if (consoleProxyUrlDomain != null && consoleProxyUrlDomain.startsWith("*")) {
```

Comment:
Can we reorder this if-then-else to:

````
if (StringUtils.isBlank(consoleProxyUrlDomain) {
   // Blank config, we use the proxy IP
   sb.append(proxyIpAddress);
} else if (consoleProxyUrlDomain.startsWith("*")) {
   sb.append(proxyIpAddress.replaceAll("\\.", "-"));
   sb.append(consoleProxyUrlDomain.substring(1)); // skip the *
} else {
   // Otherwise we assume a valid domain if config not blank
   sb.append(consoleProxyUrlDomain);
}
````